File: 11 - Movie Review System/movie-review-frontend/src/api/movieService.js
import api from "./api";

export const getAllMovies = () => api.get("/movies");

export const getMovieById = (id) => api.get(`/movies/${id}`);

export const addMovie = (movie) => api.post("/movies", movie);

export const updateMovie = (id, movie) =>
    api.put(`/movies/${id}`, movie);

export const deleteMovie = (id) =>
    api.delete(`/movies/${id}`);

export const searchMovie = (title) =>
    api.get(`/movies/search?title=${title}`);

export const filterMovie = (genre) =>
    api.get(`/movies/genre?genre=${genre}`);

File: 11 - Movie Review System/moviereview/src/main/java/com/dhanush/moviereview/service/MovieService.java
package com.dhanush.moviereview.service;

import com.dhanush.moviereview.dto.MovieRequest;
import com.dhanush.moviereview.entity.Movie;

import java.util.List;

public interface MovieService {

    Movie addMovie(MovieRequest request);

    List<Movie> getAllMovies();

    Movie getMovieById(Long id);

    Movie updateMovie(Long id, MovieRequest request);

    void deleteMovie(Long id);

    List<Movie> searchByTitle(String title);

    List<Movie> filterByGenre(String genre);

}